### DISCUSSION TITLE
Still struggling getting autoformating to work ..

### DISCUSSION THREAD

**Discussion by tf42src** [LazyVim/LazyVim#3369] [Q&A] [answered]
I have a simple html file, I have installed `formatting.prettier` using `:LazyExtras`:

![Screenshot from 2024-05-30 18-20-04](https://github.com/LazyVim/LazyVim/assets/9712861/a61dd2f9-8b74-45b7-9c11-29779b0eb383)

Running `:LazyFormat` results in the following error:

![Screenshot from 2024-05-30 18-17-46](https://github.com/LazyVim/LazyVim/assets/9712861/2f1971d2-7766-421d-bf56-3e82e35254b5)

`:ConformInfo` shows me an error but still not what the actual issue is:

![Screenshot from 2024-05-30 18-18-40](https://github.com/LazyVim/LazyVim/assets/9712861/aa7b14c7-d524-4ea8-98d8-27626f6000f4)

`:LazyFormatInfo` output:

![Screenshot from 2024-05-30 18-18-23](https://github.com/LazyVim/LazyVim/assets/9712861/57f7db28-b93a-415e-8d7e-b5002458c80c)

My filetype `:set ft`is plain html:

![Screenshot from 2024-05-30 18-21-43](https://github.com/LazyVim/LazyVim/assets/9712861/9880e75d-3879-4073-ab09-b015871c8aaf)

This is my *plugins/formatter.lua*:

```
return {
  {
    "stevearc/conform.nvim",
    opts = {
      formatters_by_ft = {
        htmldjango = { "djlint" },
        html = { "prettierd", "prettier" },
        python = { "isort", "black" },
        javascript = { { "prettierd", "prettier" } },
      },
      format = {
        timeout_ms = 30000,
      },
    },
  },
}
```

and running `prettier --write plain.html` on the console works.

```
NVIM v0.10.0
Build type: RelWithDebInfo
LuaJIT 2.1.1713484068
Run "nvim -V1 -v" for more info
```

Any ideas?

**Comment by tf42src** [comment]
This might be the issue:

![Screenshot from 2024-05-30 18-29-42](https://github.com/LazyVim/LazyVim/assets/9712861/891f2bb5-bcfd-4466-8918-e70299eb8ceb)

prettier is installed in `./node_modules/prettier` and the binary is located in `./node_modules/.bin/prettier`

**Reply by tf42src** [reply]
After adding `./node_modules/.bin` to the path in my (fish) shell config, lazyvim does find it, but I am still getting the same (empty) error as before.

**Reply by dpetka2001** [reply]
Does your file get formatted when you make a change?

**Reply by tf42src** [reply]
Nope. The first time I run `:LazyFormat` I am getting the error message and nothing afterwards anymore - until I restart nvim.

**Reply by dpetka2001** [reply]
You don't have `prettierd` executable it seems. Why do you define it in the formatters? Either remove it or add an additional pair of brackets like you do for `javascript` filetype. From conform's docs examples
> -- Conform will run multiple formatters sequentially
    go = { "goimports", "gofmt" },
    -- Use a sub-list to run only the first available formatter
    javascript = { { "prettierd", "prettier" } },

**Reply by tf42src** [reply]
I tried both: `html = { { "prettierd", "prettier" } },` and `html = { "prettier" },`, restarted (lazy)nvim, but the result is still the same.

**Reply by dpetka2001** [reply]
It has something to do with your $PATH i believe. When you install the Extra for prettier formatting it also installs prettier via Mason. Is there a specific reason you have prettier installed via npm in your project?

**Reply by tf42src** [reply]
I deleted the `.node_modules` and removed the extra PATH setting. Now it can't find a formatter at all anymore.

![Screenshot from 2024-05-30 19-06-31](https://github.com/LazyVim/LazyVim/assets/9712861/fee254d6-3550-43de-a8cd-c179cdea454a)

**Reply by dpetka2001** [reply]
Open Mason with keymap `<leader>cm`. Does it list `prettier` installed?

**Reply by tf42src** [reply]
It wasn't. _Why does lazy use two different package managers?_

Trying to install it gives me the next error without proper error message :(

![Screenshot from 2024-05-30 19-11-47](https://github.com/LazyVim/LazyVim/assets/9712861/35272d0e-40b7-46b3-932e-0fa79da2b0a1)

npm is here:

![Screenshot from 2024-05-30 19-14-09](https://github.com/LazyVim/LazyVim/assets/9712861/5ee65b17-d221-4c3d-a725-1cdf44d7608c)

and `npm install prettier@3.2.5` on the command line works.

```
npm --version
10.7.0
```

**Reply by tf42src** [reply]
I removed snap npm and node, tried install via apt, now the node version is too old!! :angry: 

I will keep trying ..

**Reply by dpetka2001** [reply]
There's some kind of incompatibility with how you installed `nodejs` in your OS and Mason I guess. I use [mise](https://github.com/jdx/mise) to install and manage NodeJs versions. Maybe try that. See also related issue in [Mason repo](https://github.com/williamboman/mason.nvim/issues/1452).

**Comment by tf42src** [comment, marked as answer]
Got it working using the following steps. Hope it helps someone else:

1. Deleted node and npm via snap `sudo snap remove node npm`
2. Deleted node and npm via apt `sudo apt-get remove --purge node.js npm`
3. Deleted local node_modules folder 
4. Download node binaries from [here](https://nodejs.org/en/download/prebuilt-binaries)
5. Extracted binaries with ` tar -xvf` to your home folder (or wherever you like)
6. Set path to the `bin` directory of that new folder in .bashrc, .zshrc or config.fish (whichever you use)
7. Reopen nvim `<leader>cm` and install prettier

Thanks heaps @dpetka2001 for all the quick help once more!

**Reply by dpetka2001** [reply]
Glad you got it working. I find it rather tedious for me to install the node binaries each time myself, that's why I use `mise` as I mentioned in my previous post to install and manage `NodeJs` and other similar tools. You use what is comfortable for you, I'm just suggesting `mise` as a way to make things easier for you (although there might be other tools that offer same functionality, but I'm just not aware of them).
